elsa 1.11.0 fails to compile in older Rust toolchain due to usage of the inline-const feature / update MSRV?

## Issue Description

Starting from version 1.11.0, this crate fails to compile with older Rust toolchain. This issue can be reproduced simply by running the following script:

```powershell
mkdir elsa-test
cd elsa-test
echo "nightly-2023-05-27" > rust-toolchain
cargo init
cargo add elsa
cargo c
```

Note that the toolchain version can be arbitrary before `"nightly-2024-03-xx"`. A successful compilation is expected, while the actual output is:

```
    Checking stable_deref_trait v1.2.0
    Checking elsa v1.11.0
error[E0658]: inline-const is experimental
   --> C:\Users\Nekomaru\.cargo\registry\src\index.crates.io-6f17d22bba15001f\elsa-1.11.0\src\sync.rs:716:10
    |
716 |         [const { AtomicPtr::new(std::ptr::null_mut()) }; NUM_BUFFERS]
    |          ^^^^^
    |
    = note: see issue #76001 <https://github.com/rust-lang/rust/issues/76001> for more information
    = help: add `#![feature(inline_const)]` to the crate attributes to enable

For more information about this error, try `rustc --explain E0658`.
error: could not compile `elsa` (lib) due to previous error
```

The `inline_const` feature was stabilized and merged into master in PR rust-lang/rust#76001 in April 24th 2024, and shipped in stable rust in 1.79.0 (See https://releases.rs/docs/1.79.0/), so recent stable and nightly toolchains accept this syntax without complaining or requiring a feature gate. For toolchains before that, this issue occurs, making the 1.11.0 version a breaking change for projects depending on this crate while using an older rust toolchain.

This issue does not occur in `elsa` version 1.10.0 and before.

## Suggested Solution

This issue can be handled simply by adding an MSRV requirement to this crate (`version = "1.79.0"` to `Cargo.toml`). I can raise a pull request for this change. Other alternatives might be removing all usage of the inline-const feature or bumping the major version, which is not recommended in any way.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "elsa"
-version = "1.11.0"
+version = "1.11.1"
 authors = ["Manish Goregaokar <manishsmail@gmail.com>"]
 edition = "2018"
 description = "Append-only collections for Rust where borrows to entries can outlive insertions"

--- a/src/sync.rs
+++ b/src/sync.rs
@@ -712,8 +712,10 @@ impl<T: Copy> Default for LockFreeFrozenVec<T> {
 }
 
 impl<T: Copy> LockFreeFrozenVec<T> {
+    const NULL_ATOMIC_PTR: AtomicPtr<T> = AtomicPtr::new(std::ptr::null_mut());
+
     const fn null() -> [AtomicPtr<T>; NUM_BUFFERS] {
-        [const { AtomicPtr::new(std::ptr::null_mut()) }; NUM_BUFFERS]
+        [Self::NULL_ATOMIC_PTR; NUM_BUFFERS]
     }
 
     pub const fn new() -> Self {

